src/com/xilinx/rapidwright/interchange/PhysNetlistReader.java

```diff
@@ -296,6 +296,11 @@ private static void readPlacement(PhysNetlist.Reader physNetlist, Design design,
                             strings.get(otherCell.getMultiCell()),
                             strings.get(otherCell.getMultiType())));
                 }else {
+                    if(c.getBEL().getPin(belPinName) == null) {
+                        System.err.println("WARNING: On cell " + c.getName() + ", a logical pin '" +
+                                c.getType() + "." + cellPinName + "' is being mapped on to a BEL pin '" 
+                                + c.getBELName() + "." + belPinName + "' that does not exist.");
+                    }

```

Comment:
My initial intent of this change is to inform those created Interchange-based designs of all the things that might be wrong (rather than spitting them out one at a time after each fix).  I think this is probably a balancing act that even Vivado plays (limiting the same kind of message after 100 occurrences).  

My other hesitancy is that I think there might be some rare cases where the BEL pin could be missing legitimately from the device model.  I'd like to avoid failing or omitting those cases as well.

Given those two constraints, what would you recommend?

Reply:
I think that's a fair reason.

My line of thinking was that this is a fix for #504 (the only being to fail more gracefully) where a NPE is triggered. With this PR, I think the understanding now is that this is some relief for that issue which doesn't directly fix the NPE but does give you a warning as to what caused it.

Reply:
Perhaps the warning message should indicate the potential impact too? At the risk of making that message even longer, maybe append "This may result in an invalid design.".